### PR TITLE
[url_launcher] Launches only mailto urls in same window on iOS devices

### PR DESCRIPTION
**Pull request by tugorez**
## Description 
Previously on https://github.com/flutter/plugins/pull/2656 we started launching all urls on the _top context (only on iOS PWA). It seems that this was a partial fix because non-PWA web apps are having a similar issue: after opening a mailto url and trying to get back to the app's context they are seeing a blank window. This behaviour is caused by the browser that opens a new tab, detects the url is one that could be handled by an email application and finally opening the email application without closing that new opened tab (leaving it with blank content). This PR assumes three things:

1. This problem only happens on iOS devices
1. This problem only happens with mailto urls
1. This problem is fixed by forcing the browser to open the mailto urls on the current window, on the _top context.

## Related Issues

flutter/flutter#51461

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/url_launcher/url_launcher_web/lib/url_launcher_web.dart [hunk 1]
```diff
@@ -4,12 +4,12 @@ import 'dart:html' as html;
 import 'package:flutter_web_plugins/flutter_web_plugins.dart';
 import 'package:meta/meta.dart';
 import 'package:url_launcher_platform_interface/url_launcher_platform_interface.dart';
-import 'src/navigator.dart' as navigator;
 
 /// The web implementation of [UrlLauncherPlatform].
 ///
 /// This class implements the `package:url_launcher` functionality for the web.
 class UrlLauncherPlugin extends UrlLauncherPlatform {
+  static final _iosPlatforms = ['iPhone', 'iPad', 'iPod'];
```
Comment (tugorez) [inline packages/url_launcher/url_launcher_web/lib/url_launcher_web.dart:12]: Could we do the same validation with https://api.flutter.dev/flutter/foundation/TargetPlatform-class.html?

Comment (tugorez) [reply packages/url_launcher/url_launcher_web/lib/url_launcher_web.dart:12]: We can use a set

**Comment by tugorez** [comment]
Please take in account that this issue seems to be happening only on iOS therefore I tried to change the behaviour only on those devices. To achieve that I had to rely on `window.navigator.platform`. My only concern is that I couldn't find an standar nor an official resource that listed all posible values. The best resource I could find was [this stackoverflow answer][1].

[1]: https://stackoverflow.com/questions/19877924/what-is-the-list-of-possible-values-for-navigator-platform-as-of-today

**Comment by ditman** [comment]
(I've sent tugorez some changes in a separate PR to this branch)

**Pull request merged**